compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp

```diff
@@ -163,7 +163,7 @@ namespace cppmicroservices
         {
             bool allConfigsAvailable = configProperties.size() >= metadata->configurationPids.size();
 
-            if ((metadata->configurationPolicy != CONFIG_POLICY_REQUIRE) || (allConfigsAvailable))
+            if (((metadata->configurationPolicy != CONFIG_POLICY_REQUIRE) || (allConfigsAvailable)) && metadata->factoryComponentID.empty())
```

Comment:
there should be a comment explaining why we don't register

Reply:
resolved